Shoko.CLI/Worker.cs

```diff
@@ -4,73 +4,25 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Hosting;
-using NLog;
 using Shoko.Server.Commands;
 using Shoko.Server.Server;
-using Shoko.Server.Settings;
 using Shoko.Server.Utilities;
 #endregion
 namespace Shoko.CLI;
 
 public class Worker : BackgroundService
 {
-    private static readonly Logger                    Logger = LogManager.GetCurrentClassLogger();
     private readonly        IHostApplicationLifetime? _appLifetime;
-
     public Worker() { }
     public Worker(IHostApplicationLifetime lifetime) => _appLifetime = lifetime;
 
     protected override Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        SetInstanceIfNeeded();
-        InitInstanceLogger();
-        LoadSettings();
-        if (ServerCouldNotStart())
-            return Task.CompletedTask;
-        EnsureAniDBSocketInitialized();
-        AddEventHandlers();
+        StartServer.StartupServer(AddEventHandlers, ServerCouldStart);
```

Comment:
Out of curiosity why refactor out this? If you're wanting it testable, you would need to have some sort of DI, but it is simply a static class which adds little to no benefit, given there isn't a huge amount of logic in the worker.

Reply:
It removes duplicate code from CLI and TrayService, and merges that code into a smaller, easier to handle shape.

> If you're wanting it testable, you would need to have some sort of DI

wdym by that? You can literally just call it in a Testproject.

Reply:
If you wanted to test Worker.cs, you can't without a hard dependency to StartServer.

Reply:
But you already have a hard dependency on the Server Project from the Worker, simple because it starts the server? What exactly is the issue with that?